packages/contracts/contracts/discovery/Batch.sol

```diff
@@ -4,32 +4,277 @@ pragma solidity =0.8.12;
 import {IBatch} from "./IBatch.sol";
 import {ICommon} from "./ICommon.sol";
 
+import {Math} from "../libraries/Math.sol";
+
+import "hardhat/console.sol";
+
 contract Batch is IBatch, ICommon {
-    uint256 batchId;
-    /// Ids of the projects
-    uint256[] projectIds;
-    /// number of approved projects in this batch
-    uint256 projectCount;
+    enum ProjectStatus {
+        InProgress,
+        Won,
+        Lost
+    }
+
+    struct Project {
+        address projectAddress;
+        ProjectStatus status;
+    }
+
+    struct Vote {
+        uint256 originalIndex;
+        uint32 count;
+    }
+
+    /// List of addresses for the project in the batch
+    address[] public projectAddresses;
+
+    /// Id of the batch
+    uint256 public batchId;
```

Comment:
do we still need this id?

Reply:
No idea what it's for. It was sent in the constructor.

Reply:
We don't need it. It was from the previous version